### DISCUSSION TITLE
Why is FileDialog.Filter property nullable and has a setter?

### DISCUSSION THREAD

**Discussion by MihaMarkic** [AvaloniaUI/Avalonia#8507] [Ideas]
IMO FileDialog.Filters property should be non-nullable and it should have a (public) setter. It even assigns a default list by default.

**Comment by maxkatz6** [comment]
I don't see any reason to has nullable attribute AND a default value. 

But in the new API filter is nullable and has not a default value - https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Platform/Storage/FilePickerOpenOptions.cs#L18
Same about file type details https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Platform/Storage/FilePickerFileType.cs#L26

**Reply by MihaMarkic** [reply]
> I don't see any reason to has nullable attribute AND a default value.

Yea, but what for? It'd make sense only if somebody would want to set it to null, but since there is a default value allocated, why would one want to?
Plus you never know then whether it is really null or not and when using it, you have to check it somehow (or blindly allocate a new instance of List<T>).

> But in the new API filter is nullable and has not a default value
That makes sense.
